aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -373,6 +373,34 @@ Tensor& logit_(Tensor& self, c10::optional<double> eps) {
   return at::logit_out(self, self, eps);
 }
 
+Tensor& nan_to_num_out(
+    Tensor& result,
+    const Tensor& self,
+    c10::optional<double> nan,
+    c10::optional<double> pos_inf,
+    c10::optional<double> neg_inf) {
+  auto iter = TensorIterator::unary_op(result, self);
```

Comment:
cc @ezyang @anjali411 this is actually another case where we could consider returning self (for integer tensors)